.github/workflows/upgrade-test.yml

```diff
@@ -0,0 +1,147 @@
+name: DSP Install and Upgrade
+
+on:
+  workflow_dispatch:
+    inputs:
+      released-version:
+        default: 'v1.y.y'
+        description: 'Most recently released DSP version'
+        required: true
+      candidate-version:
+        default: 'v1.x.x'
+        description: 'Candidate version for upgrade'
+        required: true
+
+concurrency:
+  group: ${{ github.head_ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  kubernetes-olm-upgrade:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 60
+    env:
+      OLM_VERSION: v0.24.0
+      VERSION: "v0.0.0-ghaction"  # Need to supply some semver version for bundle to be properly generated
+      CATALOG_BASE_IMG: "registry.access.redhat.com/redhat/community-operator-index:v4.13"
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # fetching also previous commits to get tags
+
+      - name: Setup and start KinD cluster
+        uses: ./.github/actions/kind
+
+      - name: Install OLM
+        run: |
+          kubectl create -f https://github.com/operator-framework/operator-lifecycle-manager/releases/download/${OLM_VERSION}/crds.yaml
+          # wait for a while to be sure CRDs are installed
+          sleep 1
+          kubectl create -f https://github.com/operator-framework/operator-lifecycle-manager/releases/download/${OLM_VERSION}/olm.yaml
+          echo Wait for default CatalogSource to start
+          kubectl wait -n ${{ env.CATALOG_SOURCE_NAMESPACE }} catalogsource/${{ env.CATALOG_SOURCE_NAME }} --for=jsonpath='{.status.connectionState.lastObservedState}'=READY --timeout=180s
+        env:
+          CATALOG_SOURCE_NAME: "operatorhubio-catalog"
+          CATALOG_SOURCE_NAMESPACE: "olm"
+
+      - name: Create openshift-operator namespace and OperatorGroup
+        run: |
+          kubectl create namespace openshift-operators
+          kubectl create -f .github/resources-olm-install/operatorgroup.yaml
+
+      - name: Apply CRDs
+        run: |
+          kubectl apply -f .github/resources-crds
+
+      - name: Deploy latest released OpenDataHub operator from OLM
+        id: deploy
+        run: |
+          echo Deploying  operator using Subscription
+          envsubst < .github/resources-olm-install/catalogsource.yaml > ${{ env.TEMP_DIR }}/catalogsource.yaml
```

Comment:
I am not seeing `TEMP_DIR` being defined anywhere for this job, where is it coming from? is this a gh action internal? I don't see it mentioned here https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables 

It seems like one of the steps defines it as well [here](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/296/files#diff-0c5ad036b2b8cd2db3ed561cf14418fe4da667be18cafdc3167605ed3de8f44dR10), but that would be scoped to the step. Is this being properly substituted?

Reply:
The `TEMP_DIR` is being defined right where you've pointed out, it gets stored as an environment variable [here](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/296/files#diff-0c5ad036b2b8cd2db3ed561cf14418fe4da667be18cafdc3167605ed3de8f44dR12), so the scope is GH action wide I believe. @sutaakar could you confirm please?

You can find the `TEMP_DIR` path being correctly substituted in this run on my fork: https://github.com/DharmitD/data-science-pipelines-operator/actions/runs/6113169351/job/16592134955

Reply:
ah yeah I missed the part where they pass it to `>> $GITHUB_ENV`, sounds good.